engines/sci/engine/klists.cpp

```diff
@@ -813,6 +813,9 @@ reg_t kArrayGetSize(EngineState *s, int argc, reg_t *argv) {
 }
 
 reg_t kArrayGetElement(EngineState *s, int argc, reg_t *argv) {
+	if (s->_segMan->getSegmentType(argv[0].getSegment()) == SEG_TYPE_SCRIPT)
+		return kStrAt(s, 2, argv);
```

Comment:
Fair enough, this change was the most controversial of all. This behavior is indeed strange, which is why I made a pull request out of it in the first place. Anyway, I'll leave that out